spec/features/course/homepage_spec.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.feature 'Course: Homepage' do
+  let(:instance) { create(:instance) }
+
+  with_tenant(:instance) do
+    let(:course) { create(:course, :opened) }
+    let(:course_student) { create(:course_student, :approved, course: course) }
+    let!(:valid_announcement) { create(:course_announcement, course: course) }
+    before do
+      login_as(user, scope: :user)
+      visit course_path(course)
+    end
+
+    context 'As a user registered for the course' do
+      let(:user) { course_student.user }
```

Comment:
This removes the need for course_student, since you can create both here and return the user?